apps/examples/security_test/hal/hal_ss_test.c

```diff
@@ -0,0 +1,162 @@
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <tinyara/security_hal.h>
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <pthread.h>
+
+#include <stress_tool/st_perf.h>
+#include "hal_test_utils.h"
+#include "hal_test.h"
+
+/*
+ * Desc: Write data in secure storage
+ * Refered https://developer.artik.io/documentation/security-api/see-securestorage-test_8c-example.html
+ */
+static hal_data g_input;
+static hal_data g_output[HAL_TEST_MAX_SLOT_INDEX];
+TEST_SETUP(write_storage)
+{
+	ST_START_TEST;
+
+	unsigned char input[HAL_TEST_MAX_DATA];
+	memset(input, 1, HAL_TEST_MAX_DATA);
+	g_input.data = input;
+	g_input.data_len = sizeof(input);
+
+	ST_END_TEST;
+}
+
+TEST_TEARDOWN(write_storage)
+{
+	ST_START_TEST;
+
+	for (uint32_t i = 0; i < HAL_TEST_MAX_SLOT_INDEX; i++) {
+		ST_EXPECT_EQ2(HAL_SUCCESS, HAL_NOT_SUPPORTED, hal_delete_storage(i));
+	}
+
+	ST_END_TEST;
+}
+
+TEST_F(write_storage)
+{
+	ST_START_TEST;
+
+	for (uint32_t i = 0; i < HAL_TEST_MAX_SLOT_INDEX; i++) {
+		ST_EXPECT_EQ2(HAL_SUCCESS, HAL_NOT_SUPPORTED, hal_write_storage(i, &g_input));
+	}
+
+	ST_END_TEST;
+}
+
+/*
+ * Desc: Read data in secure storage
+ * Refered https://developer.artik.io/documentation/security-api/see-securestorage-test_8c-example.html
+ */
+TEST_SETUP(read_storage)
+{
+	ST_START_TEST;
+
+	unsigned char input[HAL_TEST_MAX_DATA];
+	memset(input, 1, HAL_TEST_MAX_DATA);
+	g_input.data = input;
+	g_input.data_len = HAL_TEST_MAX_DATA;
+	for (uint32_t i = 0; i < HAL_TEST_MAX_SLOT_INDEX; i++) {
+		ST_EXPECT_EQ2(HAL_SUCCESS, HAL_NOT_SUPPORTED, hal_write_storage(i, &g_input));
+	}
+
+	ST_END_TEST;
+}
+
+TEST_TEARDOWN(read_storage)
+{
+	ST_START_TEST;
+
+	for (uint32_t i = 0; i < HAL_TEST_MAX_SLOT_INDEX; i++) {
+		ST_EXPECT_EQ2(HAL_SUCCESS, HAL_NOT_SUPPORTED, hal_delete_storage(i));
+		hal_test_free_buffer(&g_output[i]);
+	}
+
+	ST_END_TEST;
+}
+
+TEST_F(read_storage)
+{
+	ST_START_TEST;
+
+	for (uint32_t i = 0; i < HAL_TEST_MAX_SLOT_INDEX; i++) {
+		ST_EXPECT_EQ(0, hal_test_malloc_buffer(&g_output[i], HAL_SS_TEST_MEM_SIZE)); 
+		ST_EXPECT_EQ2(HAL_SUCCESS, HAL_NOT_SUPPORTED, hal_read_storage(i, &g_output[i]));
+	}
+
+	ST_END_TEST;
+}
+
+/*
+ * Desc: Delete data in secure storage
+ * Refered https://developer.artik.io/documentation/security-api/see-securestorage-test_8c-example.html
+ */
+TEST_SETUP(delete_storage)
+{
+	ST_START_TEST;
+
+	unsigned char input[HAL_TEST_MAX_DATA];
+	memset(input, 1, HAL_TEST_MAX_DATA);
+	g_input.data = input;
+	g_input.data_len = HAL_TEST_MAX_DATA;;
+	for (uint32_t i = 0; i < HAL_TEST_MAX_SLOT_INDEX; i++) {
+		ST_EXPECT_EQ2(HAL_SUCCESS, HAL_NOT_SUPPORTED, hal_write_storage(1, &g_input));
+	}
+
+	ST_END_TEST;
+}
+
+TEST_TEARDOWN(delete_storage)
+{
+	ST_START_TEST;
+
```

Comment:
Is this TEST_TEARDOWN call required ?

Reply:
It should be defined, following the rule of stress tool. Thanks for the comment.